src/pydvl/value/sampler.py

```diff
@@ -372,6 +382,27 @@ def weight(cls, n: int, subset_len: int) -> float:
         return n * math.comb(n - 1, subset_len) if n > 0 else 1.0
 
 
+class AntitheticPermutationSampler(PermutationSampler[IndexT]):
+    """Samples permutations like
+    [PermutationSampler][pydvl.value.sampler.PermutationSampler], but after
+    each permutation, it returns the same permutation in reverse order.
+
+    This sampler was suggested in (Mitchell et al. 2022)<sup><a
+    href="#mitchell_sampling_2022">1</a></sup>
+    """
+
+    def __iter__(self) -> Iterator[SampleT]:
+        while True:
+            permutation = self._rng.permutation(self._indices)
+            for perm in permutation, permutation[::-1]:
+                for i, idx in enumerate(perm):
+                    yield idx, perm[:i]
+                    self._n_samples += 1
```

Comment:
Shouldn't we yield the reversed permutation right after the first one?
If I'm not mistaken, this implementation goes through the entire permutation first before yielding the reversed ones.  

Perhaps this would be better?
```suggestion
            permutation = self._rng.permutation(self._indices)
            for i, idx in enumerate(permutation):
                 yield idx, permutation[:i]
                 self._n_samples += 1
                 yield idx, permutation[i+1:][::-1]
                 self._n_samples += 1
```

For the array of indices `indices = np.array([1, 2, 3])` this should generate (ignoring the permutation):

```python
1 []
1 [3, 2]
2 [1]
2 [3]
3 [1, 2]
3 []
```

Reply:
Yes, it does one full permutation before the other one. This is how it's described in the paper which first described antithetic permutation sampling. It's the same as with sets: first one set, then the complement. In this case a permutation, then its reverse. We could also start from both ends. Each of these has different properties that I haven't investigated at all, so I was content with doing what the paper describes, but all this probably warrants documenting and warning.

Reply:
See #442